homeassistant/components/enocean/dongle.py

```diff
@@ -0,0 +1,85 @@
+"""Representation of an EnOcean dongle."""
+import glob
+import logging
+from os.path import basename, normpath
+
+from enocean.communicators import SerialCommunicator
+from enocean.protocol.packet import RadioPacket
+import serial
+
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import SIGNAL_RECEIVE_MESSAGE, SIGNAL_SEND_MESSAGE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EnOceanDongle:
+    """Representation of an EnOcean dongle.
+
+    The dongle is responsible for receiving the ENOcean frames,
+    creating devices if needed, and dispatching messages to platforms.
+    """
+
+    @staticmethod
+    def detect():
+        """Return a list of candidate paths for USB ENOcean dongles.
+
+        This method is currently a bit simplistic, it may need to be
+        improved to support more configurations and OS.
+        """
+        globs_to_test = ["/dev/tty*FTOA2PV*", "/dev/serial/by-id/*EnOcean*"]
+        found_paths = []
+        for current_glob in globs_to_test:
+            found_paths.extend(glob.glob(current_glob))
+
+        return found_paths
+
+    @staticmethod
+    def validate_path(path: str):
+        """Return True if the provided path points to a valid serial port, False otherwise."""
+        try:
+            test_port = serial.Serial(path, 57600, timeout=0.1)
```

Comment:
Could we make a `SerialCommunicator` instance instead and try the same concept?

Reply:
SerialCommunicator is a subclass of Thread, I was thinking that starting a thread for that is a bit overkill.
But indeed, I can use SerialCommunicator.

Reply:
Change done